examples/concatenate.py

```diff
@@ -15,16 +15,17 @@ def concat2d_dim1(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
     )
     for tile0, tile1 in hl.tile(out.size()):
         # Most masking is automatic in helion, but tile1 spans both x and y we need to do some manual masking
+        tile1_indices = hl.tile_index(tile1)
         x_part = hl.load(
-            x, [tile0, tile1], extra_mask=(tile1.index < x.size(1))[None, :]
```

Comment:
If we treat `tile` has a Python slice object, then `tile.index` no longer works.

To make the UX better, I am thinking of adding a `RefTile` class that Dynamo can understand, and support tile APIs like `.index` / `.begin` / `.end` in that class.